build/Common.props

```diff
@@ -36,11 +36,12 @@
     <MicrosoftCodeAnalysisAnalyzersPkgVer>[3.3.3]</MicrosoftCodeAnalysisAnalyzersPkgVer>
     <MicrosoftCodeCoveragePkgVer>[17.4.1]</MicrosoftCodeCoveragePkgVer>
     <MicrosoftExtensionsConfigurationEnvironmentVariablesPkgVer>[3.1.0,)</MicrosoftExtensionsConfigurationEnvironmentVariablesPkgVer>
-    <MicrosoftExtensionsDependencyInjectionAbstractionsPkgVer>[3.1.0,)</MicrosoftExtensionsDependencyInjectionAbstractionsPkgVer>
+    <MicrosoftExtensionsDependencyInjectionPkgVer>[6.0.0,)</MicrosoftExtensionsDependencyInjectionPkgVer>
+    <MicrosoftExtensionsDependencyInjectionAbstractionsPkgVer>$(MicrosoftExtensionsDependencyInjectionPkgVer)</MicrosoftExtensionsDependencyInjectionAbstractionsPkgVer>
     <MicrosoftExtensionsHostingAbstractionsPkgVer>[2.1.0,)</MicrosoftExtensionsHostingAbstractionsPkgVer>
     <MicrosoftExtensionsLoggingPkgVer>[3.1.0,)</MicrosoftExtensionsLoggingPkgVer>
-    <MicrosoftExtensionsLoggingConfigurationPkgVer>[3.1.0,)</MicrosoftExtensionsLoggingConfigurationPkgVer>
-    <MicrosoftExtensionsOptionsPkgVer>[5.0.0,)</MicrosoftExtensionsOptionsPkgVer>
+    <MicrosoftExtensionsLoggingConfigurationPkgVer>$(MicrosoftExtensionsLoggingPkgVer)</MicrosoftExtensionsLoggingConfigurationPkgVer>
+    <MicrosoftExtensionsOptionsPkgVer>[6.0.0,)</MicrosoftExtensionsOptionsPkgVer>
```

Comment:
I recall there was a reason we needed 5+, but can you remind me why we couldn't have used 3.1.0?

Reply:
We use this [CreateInstance](https://learn.microsoft.com/dotnet/api/microsoft.extensions.options.optionsfactory-1.createinstance) method (which was added in 5.0) to fire [delegates when options are retrieved](https://github.com/open-telemetry/opentelemetry-dotnet/blob/00314ac693d6e3efe55a13361a2e5e7e103595da/src/OpenTelemetry/Internal/ConfigurationExtensions.cs#L181) which is used for the [retrieve-environment-variables-from-iconfiguration feature](https://github.com/open-telemetry/opentelemetry-dotnet/blob/00314ac693d6e3efe55a13361a2e5e7e103595da/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpMetricExporterExtensions.cs#L70).